transaction/core/src/signer/dummy_signer.rs

```diff
@@ -0,0 +1,59 @@
+use super::{Error, OneTimeKeyOrAlternative, RingSigner, SignableInputRing};
+use crate::ring_signature::{generators, CryptoRngCore, RingMLSAG, Scalar};
+use mc_crypto_keys::RistrettoPublic;
+
+/// An implementation of RingSigner that holds nothing
+///
+/// This version only works if the input secret actually includes the one-time
+/// private key, and returns an error if only the alternative is supplied.
+///
+/// The purpose of this is to avoid the need to refactor existing working
+/// software like SDKs that would not benefit from migrating to the
+/// LocalRingSigner, at least for now
+#[derive(Clone, Debug)]
+pub struct DummyRingSigner {}
```

Comment:
Yeah it is doing signing, but it's a "dummy" in that it doesn't hold private keys itself and doesn't derive the one-time private key itself.

Idk, I don't get the `HardcodedOneTimeKeyRingSigner` name, because it sounds like a key is hard-coded or something, but that's not the case. I'm not sure I have a suggestion for a better name though

Reply:
Neither do I and I don't feel strongly about naming things, so it can stay as is.

Reply:
How about `SimpleRingSigner` or `PassThroughRingSigner`?

Reply:
SimpleRingSigner is what I might go for, although honestly there may a better name that expresses "use this when you're providing absolutely all the key material"

Reply:
how do you guys feel about `NoKeysRingSigner` ?

I like `PassThroughRingSigner` but i was worried it might be confused for a ring signer that sends ring signing requests to a remote device

Reply:
or, `NoDeriveRingSigner` ?

Reply:
I think `NoKeysRingSigner` is the best name. It describes EXACTLY what it is doing.